### DISCUSSION TITLE
Bug with many 2 any (M2A)

### DISCUSSION THREAD

**Discussion by iqraaaswad** [directus/directus#8118] [Archive]
I found a bug when using M2A in my collection. i have several collection that connected using single field M2A
i need to limiting every collection using RestAPI. i got correct data using graphQL.. but got wrong one using restAPI

here is my graphqlCode :
![image](https://user-images.githubusercontent.com/58161890/133810524-d00f1df9-e867-4dda-a463-577bfd57dcef.png)

my restAPI code : 
`127.0.0.1:8055/items/homepage?fields=section.item.*&deep[section][item][products][_limit]=1`

`deep[section][item][products][_limit]=1` seems doesnt work for me.. someone can help me pls ? or provide me some explanation. tq before

**Comment by rijkvanzanten** [comment]
I don't have your exact schema configured, so I haven't tested this, but you should be a able to rewrite your GraphQL query as follows:

```graphql
query {
  homepage {
    section {
       item {
         ... on categories {
           products(limit: 1) {
             products_id {
               name
             }
           }
         }
       }
    }
  }
}
```

```
GET /items/homepage
   ?fields=section.item:categories.products.products_id.name
   &deep[section][item:categories][products][_limit]=1
```

See https://docs.directus.io/reference/api/query/#many-to-any-union-types on how to work with Many to Any fields in the REST API query parameters.

**Reply by iqraaaswad** [reply]
thanks dude.. work perfectly ^^
